django/test/client.py

```diff
@@ -331,6 +343,8 @@ def get(self, path, data=None, secure=False, **extra):
     def post(self, path, data=None, content_type=MULTIPART_CONTENT,
              secure=False, **extra):
         """Construct a POST request."""
+        if content_type == JSON_CONTENT:
```

Comment:
It might be better to put the content type check in _encode_json for consistency with _encode_data and to avoid duplicating that in every put/patch/etc. method.

Reply:
Pointed this out on the ticket, my only issue with it is I don't think it's clear that this check will happen with this method name.